ignite/engine/engine.py

```diff
@@ -800,6 +818,8 @@ def _run_once_on_dataset(self) -> float:
                     if self.state.epoch_length is None:
                         # Define epoch length and stop the epoch
                         self.state.epoch_length = iter_counter
+                        if self.state.max_iters is None:
```

Comment:
I don't understand why `max_iters` is initialized if `None`

Reply:
Let's do the assignement according to 5.b) case described above